### PR TITLE
feat(file): default icon sizes are now available for new image thumbs

### LINKED ISSUE #6181: File plugin should be using default icon_sizes
The file plugin uses its own icon_sizes. I think it should be nice if it uses also the default icon_sizes array, so themers have all thumbnail sizes available when changing look and feel of the file plugin. If the extra sizes for file are still required, the could easily be added.

Unexpected behaviour by using other thumbnail generation example here #6168

This change could best be combined with https://github.com/Elgg/Elgg/issues/6152 so there is no issue when introducing new icon_sizes for files that already have thumbnails generated.

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #6181

ref #3907

### CODE REVIEW

File: mod/file/actions/file/upload.php [hunk 1]
```diff
@@ -51,46 +50,27 @@
...
 $file->tags = string_to_tag_array($tags);
 
+$file->save();
+
 if ($uploaded_file && $uploaded_file->isValid()) {
-	if ($file->acceptUploadedFile($uploaded_file)) {
-		$guid = $file->save();
+	// save master file
+	$file->acceptUploadedFile($uploaded_file);
+	if (!$file->save()) {
```
Comment (jeabakker) [inline mod/file/actions/file/upload.php]: is there the potential that this fails here? Shouldn't this check the result of `$file->acceptUploadedFile()`

File: mod/file/start.php [hunk 2]
```diff
@@ -479,11 +479,6 @@ function file_set_icon_file($hook, $type, $icon, $params) {
 			$metadata_name = 'smallthumb';
 			break;
 
-		case 'large' :
```
Comment (jeabakker) [inline mod/file/start.php]: why only remove `large`?

**Review by jeabakker — DISMISSED** [review]

File: mod/file/start.php (hunk 2 above)
Comment (jdalsem) [reply mod/file/start.php]: behaviour of large is the same as default... other cases make different names

File: mod/file/actions/file/upload.php (hunk 1 above)
Comment (jdalsem) [reply mod/file/actions/file/upload.php]: yes

**Pull request merged**